node_modules/gh-orgunit/tests/test-import.js

```diff
@@ -0,0 +1,806 @@
+                            });
+                        });
+                    });
+                });
+            });
+        });
+
+        /**
+         * Test that verifies that external ids have to be unique
+         */
+        it('verify external ids have to be unique within the tree', function(callback) {
+            OrgUnitTestsUtil.assertCreateOrgUnit(global.tests.admins.cam2013.client, 'Test part', 'part', {}, function(part) {
+
+                // Trying to import 2 organisatioanl units with the same external id should fail
+                var importData = _getTestTree('1part-2module-1series-1event');
+                importData.children[1].externalId = importData.children[0].externalId;
+                OrgUnitTestsUtil.assertImportOrgUnitImportFails(global.tests.admins.cam2013.client, part.id, importData, false, ['Duplicate externalId'], function() {
+
+                    // Trying to import an organisational unit with the same external id as a series should fail
+                    importData = _getTestTree('1part-1module-1series-1event');
+                    importData.children[0].externalId = importData.children[0].series[0].externalId;
+                    OrgUnitTestsUtil.assertImportOrgUnitImportFails(global.tests.admins.cam2013.client, part.id, importData, false, ['Duplicate externalId'], function() {
+
+                        // Trying to import an organisational unit with the same external id as an event should fail
+                        importData = _getTestTree('1part-1module-1series-1event');
+                        importData.children[0].externalId = importData.children[0].series[0].events[0].externalId;
+                        OrgUnitTestsUtil.assertImportOrgUnitImportFails(global.tests.admins.cam2013.client, part.id, importData, false, ['Duplicate externalId'], function() {
+
+                            // Trying to import a series with the same external id as an event should fail
+                            importData = _getTestTree('1part-1module-1series-1event');
+                            importData.children[0].series[0].externalId = importData.children[0].series[0].events[0].externalId;
+                            OrgUnitTestsUtil.assertImportOrgUnitImportFails(global.tests.admins.cam2013.client, part.id, importData, false, ['Duplicate externalId'], function() {
+
+                                // Import the structure into the part
+                                importData = _getTestTree('1part-2module-1series-1event');
+                                OrgUnitTestsUtil.assertImportOrgUnit(global.tests.admins.cam2013.client, part.id, importData, false, function() {
+
+                                    // Verify we can import the SAME structure into ANOTHER part. This is necessary
+                                    // as multiple consumers can import data, but they don't necessarily know about
+                                    // each other's external ids.
```

Comment:
Does this mean that we can't add a DB constraint to make `externalId`'s globally unique in the system?